autoload/go/guru.vim

```diff
@@ -364,6 +363,12 @@ function! go#guru#SameIds(selected)
     let pos = split(item, ':')
     call matchaddpos('goSameId', [[str2nr(pos[-2]), str2nr(pos[-1]), str2nr(poslen)]])
   endfor
+
+  if get(g:, "go_auto_sameids", 0)
+    " re-apply SameIds at the current cursor position at the time the buffer
+    " is redisplayed: e.g. :edit, :GoRename, etc.
+    autocmd BufWinEnter <buffer> nested call go#guru#SameIds(-1)
```

Comment:
That's exactly right. But the nice thing about this approach, is that it will do the right thing even the user does `:edit`.

I don't think there's a case that fails to remove the buffer-local autocommand; the only case I could think of is when the buffer is wiped out, but in that case the buffer-local autocommand will be removed by Vim.